### PR TITLE
[RHOAIENG-5073] - Routing and Headless Service Support in KServe Raw Mode Deployment

### PR DESCRIPTION
**Pull request by spolti**
Cherry-picks https://github.com/opendatahub-io/kserve/pull/280

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -325,9 +325,10 @@ kustomize: $(KUSTOMIZE) ## Download kustomize locally if necessary.
 $(KUSTOMIZE): $(LOCALBIN)
 	test -s $(LOCALBIN)/kustomize || { curl -Ss $(KUSTOMIZE_INSTALL_SCRIPT) | bash -s -- $(subst v,,$(KUSTOMIZE_VERSION)) $(LOCALBIN); }
 
+ENVTEST_PACKAGE_VERSION = v0.0.0-20240320141353-395cfc7486e6
 envtest: $(ENVTEST) ## Download envtest-setup locally if necessary.
 $(ENVTEST): $(LOCALBIN)
-	test -s $(LOCALBIN)/setup-envtest || GOBIN=$(LOCALBIN) go install sigs.k8s.io/controller-runtime/tools/setup-envtest@latest
+	test -s $(LOCALBIN)/setup-envtest || GOBIN=$(LOCALBIN) go install sigs.k8s.io/controller-runtime/tools/setup-envtest@$(ENVTEST_PACKAGE_VERSION)
```
Comment (terrytangyuan) [inline Makefile:331]: Unnecessary change

Comment (spolti) [reply Makefile:331]: without it CI fails, same as here:
https://github.com/opendatahub-io/odh-model-controller/pull/184

**Review by terrytangyuan — APPROVED** [review]
/lgtm

**Comment by spolti** [comment]
Seems that the regex is wrong here:

https://github.com/openshift/release/blob/master/ci-operator/jobs/opendatahub-io/odh-model-controller/opendatahub-io-odh-model-controller-release-0.11.1-presubmits.yaml#L6
